vcd/resource_vcd_vapp_network.go

```diff
@@ -219,43 +270,169 @@ func resourceVappNetworkRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	vAppNetwork := types.VAppNetworkConfiguration{}
+	var networkId string
 	for _, networkConfig := range vAppNetworkConfig.NetworkConfig {
-		if networkConfig.NetworkName == d.Get("name").(string) {
+		networkId, err = govcd.GetUuidFromHref(networkConfig.Link.HREF)
```

Comment:
Wondering if we can "outsource" this network ID lookup into govcd

Reply:
It outsourced already GetUuidFromHref is govcd function